### DISCUSSION TITLE
Checking Multiple Subjects or Actions in a Single casl-react Can Check

### DISCUSSION THREAD

**Discussion by nbibler** [stalniy/casl#921] [Q&A] [answered]
Is it possible to check for any one of many abilities in a single `<Can …>` from `casl-react`? I imagine this could get hairy if the actions differ, so I'm guessing the answer is "no." 😉

For example, it seems like it'd be useful to hide a list entirely if every item in that list cannot be performed by the current ability:

```tsx
<Can I="edit" a={["User", "Post"]}>
  <ul>
    <Can I="edit" a="User">
      <li><a href="users/edit">Edit Users</a></li>
    </Can>
    <Can I="edit" a="Post">
      <li><a href="posts/edit">Edit Posts</a></li>
    </Can>
  </ul>
</Can>
```

**Comment by serhiistotskyi** [comment, marked as answer]
Can you use use `useAbility` hook instead? if so, then you can just do `ability.can('edit', 'User') || ability.can('edit', 'Post')`

Your expectations is that `ability.can('edit', ['User', 'Post')` should return true if at least one subject type can be edited but somebody else may expect that both should be editable. This complicates types definitions. Then somebody else will request feature like `can(['edit', 'create'], ['Post', 'User'])` afterwards `can(['edit', 'create'], ['Post', 'User'], ['status', 'author'])`

This feature was requested many times and once I even tried to work on it but as far as I remember it quickly caused headache :)

**Reply by nbibler** [reply]
Yep, agreed. This seems like a good opportunity to use the `ability` directly and just build a custom boolean value in the component for hiding the list (in this case).

Thanks!
